### PR TITLE
Copy deploy.sh from develop

### PR DESCRIPTION
**Pull request by wolf31o2**
This includes SSH host key checking fix.

### CODE REVIEW

File: cdap-docs/deploy.sh [hunk 1]
```diff
@@ -186,6 +183,12 @@ function deploy () {
   local _version=${6}
   local _branch=${7}
   decho "deploying to ${_host}"
+  # if there is no branch name to append, append version to directory name, otherwise:
+  #   cdap dir gets blown away in remote dir delete/create
+  #   otherwise all files end up one level too high after rsync
```
Comment (awholegunch) [inline cdap-docs/deploy.sh]: otherwise twice. "all files end up one level too high after rsync"

File: cdap-docs/deploy.sh [hunk 2]
```diff
@@ -200,16 +203,20 @@ decho "DEPLOY_TO_DOCS=${DEPLOY_TO_DOCS}"
...
-  for i in ${DOCS_SERVERS}; do
-    deploy ${USER} ${i} ${REMOTE_DOCS_DIR} ${ZIP_FILE} ${FILE_PATH} ${VERSION} ${BRANCH}
-  done
+  if [ "${BRANCH}" == '' ]; then
+    decho "Deploying artifacts to Docs servers"
+    for i in ${DOCS_SERVERS}; do
+      deploy ${USER} ${i} ${REMOTE_DOCS_DIR} ${ZIP_FILE} ${FILE_PATH} ${VERSION}
+    done
+  else
+    decho "Do not deploy artifacts from feature branches to Docs servers"
```
Comment (awholegunch) [inline cdap-docs/deploy.sh]: This message looks odd to me; we do deploy artifacts from the feature branches (including release feature branches) to Docs servers. Why is this not deploying if Branch != ""?

**Comment by awholegunch** [comment]
A couple of minor comments, otherwise LGTM.

**Comment by wolf31o2** [comment]
This was a copy from the current develop version to release/2.8... I neither wrote nor modified this code. I literally checked out the latest version of the file and made a commit the correct branch.

**Pull request merged**